sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -218,7 +219,7 @@ def test_agglomerative_clustering(global_random_seed):
         # Check that we raise a TypeError on dense matrices
         clustering = AgglomerativeClustering(
             n_clusters=10,
-            connectivity=sparse.lil_matrix(connectivity.toarray()[:10, :10]),
+            connectivity=csr_container(connectivity.toarray()[:10, :10]),
```

Comment:
We can use the LIL_CONTAINERS over here to specify the precise type of matrix being used. https://github.com/scikit-learn/scikit-learn/blob/f55da6220ec4210650fd780163fc33c49f453a27/sklearn/utils/fixes.py#L48

Reply:
@OmarManzoor Thanks for letting me know. I fixed the mistake and undid the changes in test_optics.py.